Miscellaneous code cleanup

basil opened the pull request:
Various minor changes to improve readability.

CC @maxandersen

quintesse commented on src/main/java/dev/jbang/cli/BaseBuildCommand.java:
```diff
@@ -472,9 +472,7 @@ protected static void searchForMain(ScriptSource src, RunContext ctx, File tmpJa
 																			STRINGTYPE) != null)
 															.findFirst();
 
-					if (agentmain.isPresent()) {
```
I'm okay with the other commits in this PR, but personally I find this harder to read than the original code.

basil replied:
Dropped the commit

maxandersen commented on CONTRIBUTORS.md:
```diff
@@ -94,6 +94,7 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
   <tr>
     <td align="center"><a href="https://github.com/JakobFels"><img src="https://avatars.githubusercontent.com/u/4647840?v=4?s=100" width="100px;" alt=""/><br /><sub><b>Jakob Fels</b></sub></a><br /><a href="https://github.com/jbangdev/jbang/issues?q=author%3AJakobFels" title="Bug reports">🐛</a></td>
     <td align="center"><a href="http://davsclaus.com/"><img src="https://avatars.githubusercontent.com/u/477100?v=4?s=100" width="100px;" alt=""/><br /><sub><b>Claus Ibsen</b></sub></a><br /><a href="https://github.com/jbangdev/jbang/issues?q=author%3Adavsclaus" title="Bug reports">🐛</a></td>
+    <td align="center"><a href="https://github.com/basil"><sub><b>Basil Crow</b></sub></a><br /><a href="https://github.com/jbangdev/jbang/commits?author=basil" title="Code">💻</a></td>
```
Don't add to this file - that is done by a bot when merge is done.

basil replied:
Dropped

maxandersen commented on src/main/java/dev/jbang/cli/Catalog.java:
```diff
@@ -97,17 +97,15 @@ public Integer doCall() {
 					} catch (Exception ex) {
 						Util.warnMsg("Unable to read catalog " + ref + " (referenced from "
-								+ e.getValue().catalog.catalogRef + ")");
+								+ value.catalog.catalogRef + ")");
```
Why not use the method ?

basil replied:
Dropped

maxandersen commented on src/main/java/dev/jbang/dependencies/DependencyUtil.java:
```diff
@@ -88,7 +88,7 @@ public static ModularClassPath resolveDependencies(List<String> deps, List<Maven
 									.collect(Collectors.toList());
 		// And if we encountered URLs let's make sure the JitPack repo is available
-		if (!depIds.equals(deps) && !repos.stream().anyMatch(r -> REPO_JITPACK.equals(r.getUrl()))) {
+		if (!depIds.equals(deps) && repos.stream().noneMatch(r -> REPO_JITPACK.equals(r.getUrl()))) {
```
Why change to this ? The original could end early where as the new one needs to always iterate through the full list ?

basil replied:
Dropped

basil commented:
Has the feedback from [this review](https://github.com/jbangdev/jbang/pull/1158#pullrequestreview-840043384) and [this review](https://github.com/jbangdev/jbang/pull/1158#pullrequestreview-840105675) been addressed? If so, is there a reason this PR has not yet been approved?

quintesse commented:
I'm guessing the reason is that it's the holidays and we're all very busy :-)
We'll get to the PR, as soon as we can, your efforts are appreciated.

basil commented:
> I'm guessing the reason is

Do you need to guess regarding [your own review](https://github.com/jbangdev/jbang/pull/1158#pullrequestreview-840043384)?

quintesse commented:
> Do you need to guess regarding your own review?

I'm not the only one doing reviews, I'm not the one who can merge PRs and I can not yet read other people's minds :-)

maxandersen commented:
@basil Happy New Year! Your PR is not adding substantial value nor fixing any known issues, and you only submitted it due to my IntelliJ giveaway created over Xmas, right? So forgive me if I'm not in a hurry to merge and recognize this no-fix, no-urgency PR during Xmas when some of the existing 66+ contributors to jbang might still not have seen the giveaway. Have patience - it will be merged eventually.

basil commented:
> I'm not the only one doing reviews, I'm not the one who can merge PRs

But you did [review my PR](https://github.com/jbangdev/jbang/pull/1158#pullrequestreview-840043384) and provide feedback. So now that I have addressed your feedback, why not confirm that the feedback has been addressed by approving the PR?

basil commented:
> Your PR is not adding substantial value nor fixing any known issues […] So forgive me if I'm not in a hurry to merge and recognize this no-fix, no-urgency PR

Are you saying that this project does not recognize code cleanup contributions to the same degree as features and bug fixes?

quintesse reviewed (approved):
I don't see any problems with the code.
Some of the changes are nice, but mostly it's just _different_, not necessarily _better_.

quintesse commented:
> why not confirm that the feedback has been addressed by approving the PR?

Sure, done.

> Are you saying ....

All code contributions are recognized, even a single typo, as you can see in the contributor list. But some of us do this in their free time, so sometimes you just have to be patient. Asking nicely helps a lot as well. And yes, contributions are obviously prioritized, there are only so many hours in a day.

basil commented:
> Are you saying ....

I was addressing the author of [this comment](https://github.com/jbangdev/jbang/pull/1158#issuecomment-1004005064).

quintesse commented:
> I was addressing the author of this comment.

Yes, and I'm answering.

basil commented:
I would prefer to hear from @maxandersen regarding https://github.com/jbangdev/jbang/pull/1158#issuecomment-1004100849 as the author of the original comment.

maxandersen commented:
@all-contributors please add @basil for code

maxandersen commented:
Thanks for the contribution @basil! if still interested in the intellij license go complete the ask on the discussion thread.
